### PR TITLE
Add product controller and tests

### PR DESCRIPTION
**Pull request by Lcv004**

### CODE REVIEW

File: test/Controllers/ProductControllerTest.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+public class ProductControllerTest
+{
+    [Fact]
+    public void Add_OneProduct_VerifyAddCallOnce()
+    {
+        // Given
+        var product = ProductFixtures.BuildProduct();
+        var mockProductRepo = new Mock<IProductRepository>();
+        mockProductRepo.Setup(m => m.Add(It.IsAny<Product>())).Verifiable();
```
Comment (reynoldmorel) [inline test/Controllers/ProductControllerTest.cs:17]: Could you explain this?

**Review by reynoldmorel — CHANGES_REQUESTED** [review]
Could you add integration tests between your controller and your repo?

File: test/Controllers/ProductControllerTest.cs (hunk 1 above)
Comment (Lcv004) [reply test/Controllers/ProductControllerTest.cs:17]: Here I'm creating a mock of the product repository interface, which I'll use some lines below to initialize the product controller. Also I'm setting up the behaviour of the `Add(Product)` method of my mocked `IProductRepository` and setting it as "verifiable" to test later if the controller called the add method of `mockProductRepo` using `mockProductRepo.Verify()`.

**Comment by Lcv004** [comment]
Hello @reynoldmorel! I already added the integration tests.

**Review by reynoldmorel — APPROVED** [review]

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**